addons/sale_coupon/wizard/sale_coupon_apply_code.py

```diff
@@ -24,7 +24,7 @@ def process_coupon(self):
 
     def apply_coupon(self, order, coupon_code):
         error_status = {}
-        program = self.env['sale.coupon.program'].search([('promo_code', '=', coupon_code)])
+        program = self.env['sale.coupon.program']._get_program(order, coupon_code)
```

Comment:
Actually I was thinking more of a solution like this, it still looks messy IMO:
```suggestion
        program_domain = order._get_coupon_program_domain() # which would return [('website_id', 'in .. when website_sale_coupon is installed, an empty dict otherwise
        program_domain = expression.AND([program_domain, [('promo_code', '=', coupon_code)]])
        program = self.env['sale.coupon.program'].search(program_domain)
```

Reply:
I believe it is more generic to keep the overriding to a function that returns the program itself. who knows based on what shinanigans we will filter in future modules ? :)

Reply:
`sale_coupon` does not exist in master anymore 😅  and it is already handled in the new way of doing, I am more concerned about completely overriding `_get_program`